src/unix/fs.c

```diff
@@ -404,6 +404,36 @@ static ssize_t uv__fs_readlink(uv_fs_t* req) {
   return 0;
 }
 
+static ssize_t uv__fs_realpath(uv_fs_t* req) {
+  ssize_t len;
+  char* buf;
+
+  len = pathconf(req->path, _PC_PATH_MAX);
+
+  if (len == -1) {
+#if defined(PATH_MAX)
+    len = PATH_MAX;
+#else
+    len = 4096;
+#endif
+  }
+
+  buf = uv__malloc(len + 1);
+
+  if (buf == NULL) {
+    errno = ENOMEM;
+    return -1;
+  }
+
+  if (realpath(req->path, buf) == NULL) {
```

Comment:
If I understand `realpath` docs properly, if we pass `NULL` as the second parameter, it will take care of the memory allocation and return the path. In fact, the patch which I wanted to submit had just this

```c
static ssize_t uv__fs_realpath(uv_fs_t* req) {
  req->ptr = realpath(req->path, NULL);
  return req->ptr == NULL ? -1 : 0;
}
```

I am not sure if it is correct though.